File: drlg/drlg.go
package drlg

import (
	. "github.com/Des-Nerger/d2moo_drlg"
)

type AutoMapFn func(*ActiveRoomStrc)
type TownAutoMapFn func(int32, int32, int32, int32)

type Flags uint32

const Flag_OnClient, Flag_Refresh Flags = 0x01, 0x10

type ActStrc struct {
	NAct       Act
	DwInitSeed uint32
	NTownId    TownId
}

type ActiveRoomStrc struct {
}

type Strc struct {
}

func NewStrc(
	pAct *ActStrc,
	nAct Act,
	pArchive *Archive,
	nInitSeed uint32,
	nTownId TownId,
	nFlags Flags,
	pGame *GameStrc,
	nDifficulty Difficulty,
	fnAutoMap AutoMapFn,
	fnTownAutoMap TownAutoMapFn,
) *Strc {
	return &Strc{}
}

File: src/drlg/drlg.zig
const consts = @import("../consts.zig");
const game = @import("../game.zig");
const std = @import("std");
const mem = std.mem;

const Act = consts.Act;
const Difficulty = consts.Difficulty;
const TownId = consts.TownId;

pub const Flags = packed struct(u32) {
	onClient: bool,
	__: u3 = 0,
	refresh: bool,
	_: u27 = 0,
};

pub const ActStrc = struct {
	allocator: mem.Allocator,
	nAct: Act,
	dwInitSeed: u32,
	nTownId: TownId,
};

const ActiveRoomStrc = struct {};

pub const Strc = struct {
	allocator: mem.Allocator,

	pub fn init(
		pAct: *ActStrc,
		nAct: Act,
		nInitSeed: u32,
		nTownId: TownId,
		nFlags: Flags,
		pGame: *game.Strc,
		nDifficulty: Difficulty,
	) @This() {
		_ = nAct;
		_ = nInitSeed;
		_ = nTownId;
		_ = nFlags;
		_ = pGame;
		_ = nDifficulty;
		return .{
			.allocator = pAct.allocator,
		};
	}
};
